### DISCUSSION TITLE
graphql/apollo-angular/docs/data/queries

### DISCUSSION THREAD

**Discussion by giscus** [the-guild-org/apollo-angular#1873] [Docs Discussions]
# graphql/apollo-angular/docs/data/queries

Fetch data with the Apollo service

https://the-guild.dev/graphql/apollo-angular/docs/data/queries

**Comment by SimonCockx** [comment]
The explanation on `useInitialLoading` is unclear to me.

> the first result is emitted with the response from the ApolloLink execution chain.

I don't know what this means.

> In order to correct it

To correct what?

**Comment by SimonCockx** [comment]
Section "Inspecting Error States" could use an example.

**Comment by aleixsuau** [comment]
The example's link does not work: https://graphql-voter-app.herokuapp.com/

**Comment by flashcrow2000** [comment]
I have a scenario that I can't figure out. I'm using a watchQuery with a search param, and if I type slow enough to trigger an intermediate call, a race condition manifests like follows:
assume the query string is "tenant"
first query runs when the search string is "ten"
second query runs when the search string is "tenant"

second query return 3 items, which get displayed correctly
first query returns 12 items, which get displayed and overwrite the previous (correct) result.

The easiest fix that I can think of is to cancel the http request for the first query, if it is in a pending state when the second query gets triggered. Any idea I can do that from the QueryRef, or by using some options on the query?
